tests/e2e/observers/test_balances.py

```diff
@@ -70,6 +78,9 @@ def on_completed():
         # Lock returns false if times-out
         assert (lock.acquire(True, 5))
 
+        # Undo DB changes to prevent interaction with other tests
+        self.clear_db()
+
```

Comment:
Were you experiencing issues? I would expect other tests to clean the DB in their respective setups.